Automattic/jetpack
# Refactor rest-api client to be a "class" and make the module export an instance of it

### Chanes proposed

* This PR changes the singleton exported from the `rest-api` module into an instance of an internal `JetpackRestApiClient`.
* Also refactors Plans related components, reducer, and relate from the rest-api. **The placeholder and cache between tabs behaviour is maintained in this branch although with a different mechanism**.
* Removes unused file `_inc/client/components/plans/site-plan.jsx`

This is to avoid excessive reference to the `window.Initial_State` variables that carry the root url of the api and the generated nonce for it.

#### Testing instructions

1. Get to the admin page, Dashboard, 
1. Switch to the **Plans** tab. 
  1. Expect to see the plans loading or loaded.
1. Refresh the page
  1. Expect to see the plans loading showing a placeholder.
1. Switch to the **Apps** tab
1. Get back to the **Plans** tab
  1. Expect to see no delay in the plans render (There's a delay on the akismet button maybe but this behaviour is related to another component not affected by this PR)

![after](https://cloud.githubusercontent.com/assets/746152/17752264/60906688-64a1-11e6-8980-986576cf659c.gif)
